File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03546/p03546_num3_parsed.py
import sys

input = sys.stdin.readline

H, W = map(int, input().split())

C = []
for _ in range(10):
    C.append(list(map(int, input().split())))

A = []
for _ in range(H):
    A.append(list(map(int, input().split())))

# Floyd-Warshall algorithm to find the minimum cost to change each digit to 1
for k in range(10):
    for i in range(10):
        for j in range(10):
            C[i][j] = min(C[i][j], C[i][k] + C[k][j])

# Calculate the total minimum cost for all digits on the wall
total_cost = 0
for i in range(H):
    for j in range(W):
        if A[i][j] != -1:
            total_cost += C[A[i][j]][1]

print(total_cost)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03546/p03546_num3_parsed.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int H, W;
    cin >> H >> W;
    
    vector<vector<int>> c(10, vector<int>(10));
    for (int i = 0; i < 10; i++) {
        for (int j = 0; j < 10; j++) {
            cin >> c[i][j];
        }
    }
    
    vector<vector<int>> A(H, vector<int>(W));
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            cin >> A[i][j];
        }
    }
    
    // Floyd-Warshall algorithm to find the minimum cost to transform each digit into 1
    for (int k = 0; k < 10; k++) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                c[i][j] = min(c[i][j], c[i][k] + c[k][j]);
            }
        }
    }
    
    // Calculate the minimum total amount of MP required to turn every digit on the wall into 1
    int totalCost = 0;
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            if (A[i][j] != -1) {
                totalCost += c[A[i][j]][1];
            }
        }
    }
    
    cout << totalCost << endl;
    
    return 0;
}